biopet-utils/src/main/scala/nl/lumc/sasc/biopet/utils/Logging.scala

```diff
@@ -38,10 +38,19 @@ object Logging {
 
   private[biopet] val errors: ListBuffer[Exception] = ListBuffer()
 
-  def addError(error: String, debug: String = null): Unit = {
-    val msg = error + (if (debug != null && logger.isDebugEnabled) "; " + debug else "")
-    logger.error(msg)
-    errors.append(new Exception(msg))
+  def addError(error: String,
+               debug: Option[String] = None,
+               cause: Option[Exception] = None): Unit = {
+    val msg = error + debug.map("; " + _).getOrElse("")
+
+    cause match {
+      case Some(e) =>
+        logger.error(msg, e)
+        errors.append(new Exception(msg, e))
```

Comment:
Now you can give an own message instead of the exception you have get some a 3 party library, the other message is still in the stacktrace